### PR TITLE
fix: don't send param with no initialValue

### LINKED ISSUE #7648: [Bug] Error sending config params with invalid value ( empty string )
## Description

[What happened]

After the addition of `where` param in the google sheet plugin `editorConfig`, generate CRUD app for google sheets started failing because for `where` param there was no initialValue and as per current logic.

where there is no initialValue, an empty string was being sent to the backend.

#### Changes to Fix this issue: 

Do not send the param if initialValue is undefined.

### Steps to reproduce the behavior:

Add steps to reproduce this behavior, include console/network logs & screenshots

1. Go to the new Appsmith app
2. Click on a new page
3. Go to generate CRUD page.
4. Selected Datasource as google sheets data source
5. See error 

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production, release, deploy preview]

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

Fixes #7648

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

**Comment by mohanarpit** [comment]
@Rishabh-Rathod Is this PR still required after the fix in #7637 ?

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GoogleSheetForm.tsx [hunk 1]
```diff
@@ -173,8 +173,8 @@ function GoogleSheetForm(props: Props) {
           const valueConfig = configs[index + 1];
           if (keyConfig && valueConfig) {
             const key = keyConfig?.initialValue;
-            const value = valueConfig?.initialValue || "";
-            if (key) requestObject[key] = value;
+            const value = valueConfig?.initialValue;
+            if (key && value !== "undefined") requestObject[key] = value;
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GoogleSheetForm.tsx:177]: QQ: is `undefined` getting converted to a string here somehow? I guess another approach could be, adding a check as: `if (requestObject.hasOwnProperty(key))`

**Review by rishabhsaxena — DISMISSED** [review]

**Comment by Rishabh-Rathod** [comment]
> @Rishabh-Rathod Is this PR still required after the fix in #7637 ?

Yes, this will avoid generate CRUD feature to fail for params not having initialValue.

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GoogleSheetForm.tsx (hunk 1 above)
Comment (Rishabh-Rathod) [reply app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/GoogleSheetForm.tsx:177]: Yes, my bad.

**Review by rishabhsaxena — DISMISSED** [review]

**Review by nidhi-nair — APPROVED** [review]

**Comment by Nikhil-Nandagopal** [comment]
@Rishabh-Rathod can we merge this in? This seems like a big issue

**Comment by Rishabh-Rathod** [comment]
> @Rishabh-Rathod can we merge this in? This seems like a big issue

This issue isn't in the current release, only once the frontend changes in the PR go to release we would have the issue,
which is solved by the backend changes of this PR itself.

This PR is mainly to make generate page features not break in the future due to changes like adding a new param without initialValue.
@Nikhil-Nandagopal

**Review by sharat87 — APPROVED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**